service/s3/s3manager/batch.go

```diff
@@ -309,6 +364,51 @@ func (d *BatchDelete) Delete(ctx aws.Context, iter BatchDeleteIterator) error {
 	return nil
 }
 
+func initDeleteObjectsInput(o *s3.DeleteObjectInput) *s3.DeleteObjectsInput {
+	return &s3.DeleteObjectsInput{
+		Bucket:       o.Bucket,
+		MFA:          o.MFA,
+		RequestPayer: o.RequestPayer,
+		Delete:       &s3.Delete{},
+	}
+}
+
+// deleteBatch will delete a batch of items in the objects parameters.
+func deleteBatch(d *BatchDelete, input *s3.DeleteObjectsInput) []Error {
+	errs := []Error{}
+
+	if result, err := d.Client.DeleteObjects(input); err != nil {
+		for i := 0; i < len(input.Delete.Objects); i++ {
+			errs = append(errs, newError(err, input.Bucket, input.Delete.Objects[i].Key))
+		}
+	} else if len(result.Errors) > 0 {
+	}
+
+	return errs
+}
+
+func hasParity(o1 *s3.DeleteObjectsInput, o2 BatchDeleteObject) bool {
+	if o1.Bucket != nil && o2.Object.Bucket != nil {
+		if *o1.Bucket != *o2.Object.Bucket {
+			return false
+		}
+	}
+
+	if o1.MFA != nil && o2.Object.MFA != nil {
```

Comment:
I don't think `hasPairty` will return the correct value if `o1.MFA` is not nil but `o2.Object.MFA` is nil. It looks like this will ignore the differing values. Is this correct?

I suggest something like the following to simplify this.

```go
if aws.StringValue(o1.MFA) != aws.StringValue(o2.Object.MFA) {
    return false
}
```